Be158

Przemyslaw-Mazurek opened the pull request:
Dodanie testów jednostkowych dla LocationMapper i LocationMapper.

Commit:
Feat: add mapper for Location

Creating two methods which first maps from entity Location to LocationDto and from LocationDto to Entity Location

Commit:
Adding tests which checks locationmapper maps correctly, Tests passed

Commit:
Feat Create package named LocationTests with LocationMapper tests

Commit:
Merge branch 'develop' into BE158

# Conflicts:
#	src/main/java/pl/envelo/moovelo/entity/Location.java
#	src/test/java/pl/envelo/moovelo/MooveloApplicationTests.java

Commit:
Fix type

Commit:
Refactoring: Changed LocationTest Package

Commit:
Merge branch 'develop' into BE158

# Conflicts:
#	src/main/java/pl/envelo/moovelo/entity/Location.java
#	src/main/java/pl/envelo/moovelo/entity/events/EventInfo.java

Commit:
Fix conflicts

mdziewiur commented on src/main/java/pl/envelo/moovelo/controller/dto/location/LocationDto.java:
```diff
@@ -2,11 +2,17 @@
 public class LocationDto {
     private long id;
+
+    private List<EventInfo> eventsInfos;
```
Tutaj powinna być lista jakichś EventInfoDto, w obiekcie dto są zagnieżdżone inne obiekty dto, ewentualnie można spłaszczyć strukturę, jeśli nie potrzebujesz dużo danych

mdziewiur commented on src/main/java/pl/envelo/moovelo/controller/mapper/LocationMapper.java:
```diff
@@ -0,0 +1,45 @@
+@AllArgsConstructor
+public class LocationMapper {
+
+    public static Location mapFromLocationDtoToLocationEntity(LocationDto locationDto) {
```
1. w metodach  mapujących nie ma odniesienia do pola eventInfos, więc przy edycji obiektu Location stracimy odniesienie do wszystkich EventInfo, które już w tej lokalizacji są zapisane. 

2. W aplikacji zakładamy możliwość edycji Location, więc jako parametr metody należałoby dodać `long id`, żeby jako pierwsze ustawić przez setter id encji, które będziemy przekazywać z kontrolera/serwisu

Commit:
Merge branch 'develop' into BE158

Commit:
Fixing

Removing list of eventInfo from LocationDto and adding one parameter id to method which maps dto to entity
